src/Compilers/Core/CodeAnalysisTest/AsyncQueueTests.cs

```diff
@@ -170,41 +170,31 @@ public async Task DequeueAsyncWithCancellation()
         }
 
         [Fact]
-        public async Task DequeueAsyncWithCancellationAfterComplete()
+        public async Task EnqueueAfterDequeueAsyncWithCancellation()
         {
             var queue = new AsyncQueue<int>();
             var cts = new CancellationTokenSource();
             var task = queue.DequeueAsync(cts.Token);
-            Assert.False(task.IsCompleted);
-            queue.Enqueue(42);
-            await task.ConfigureAwait(false);
+            Assert.False(task.IsCanceled);
             cts.Cancel();
+            await Assert.ThrowsAsync<TaskCanceledException>(() => task);
+            Assert.Equal(TaskStatus.Canceled, task.Status);
+
+            queue.Enqueue(1);
+            Assert.True(queue.TryDequeue(out var value));
+            Assert.Equal(1, value);
         }
 
         [Fact]
-        public async Task TaskCompletesAsyncWithComplete()
```

Comment:
@jaredpar as best I can tell, the change in behavior that impacted this test will not impact `NamedPipeClientConnectionHost`.

Reply:
Agree I think this is is fine for our usage in the server.

Reply:
➡️ No action taken 🎉